bigchaindb/common/transaction.py

```diff
@@ -1075,14 +1188,19 @@ def gen_tx(fulfillment, condition, input_condition_uri=None):
                                                   tx_serialized,
                                                   input_condition_uri)
 
-        if not fulfillments_count == conditions_count == \
-           input_condition_uris_count:
-            raise ValueError('Fulfillments, conditions and '
+        # TODO: Why?? Need to ask @TimDaub
+        # if not fulfillments_count == conditions_count == \
+        #    input_condition_uris_count:
+        #     raise ValueError('Fulfillments, conditions and '
+        #                      'input_condition_uris must have the same count')
+        # else:
+        if not fulfillments_count == input_condition_uris_count:
+            raise ValueError('Fulfillments and '
                              'input_condition_uris must have the same count')
         else:
             partial_transactions = map(gen_tx, self.fulfillments,
                                        self.conditions, input_condition_uris)
-            return all(partial_transactions)
```

Comment:
I removed the indent. Looking at it again there is actually no reason for the `else`
It either raises an exception or returns `all(partial_transactions)`